### PR TITLE
Close #12174: Remove Formatter additions with uintptr_t

### LINKED ISSUE #12174: Remove Formatter additions with uintptr_t
The `Formatter` class has a method `Add` for writing to a format string. Some of the calls to this method are using aliased types, not their actual types.

This issue is to remove the `Add<uintptr_t>` calls and instead pass the type they want to be written (like `const char*` or `uint8_t`)

### PR DESCRIPTION
**Pull request by heliobatimarqui**

### CODE REVIEW

**Comment by Gymnasiast** [comment]
There was already a PR for this: https://github.com/OpenRCT2/OpenRCT2/pull/12199

I'm therefore closing this one.

**Comment by Gymnasiast** [comment]
There was a PR to address #12174 already: https://github.com/OpenRCT2/OpenRCT2/pull/12199
I'm therefore closing this one.

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 1]
```diff
@@ -717,7 +717,7 @@ static void window_loadsave_paint(rct_window* w, rct_drawpixelinfo* dpi)
 
     // Draw path text
     auto ft = Formatter::Common();
-    ft.Add<uintptr_t>(Platform::StrDecompToPrecomp(buffer));
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/LoadSave.cpp:720]: Why did you remove this call? It is there for a good reason.

Comment (heliobatimarqui) [reply src/openrct2-ui/windows/LoadSave.cpp:720]: Hi. I've seen that you closed it now. I didn't see that there was a PR for this already. In regard of removing the call, if I recall correctly this funtion is purely a cast.

Comment (Gymnasiast) [reply src/openrct2-ui/windows/LoadSave.cpp:720]: Note that it lives in `Platform`. It means the implementation is OS-dependent. Take a look at the macOS implementation and you will see that is not purely a cast.
